olcut-core/src/main/java/com/oracle/labs/mlrg/olcut/config/ConfigurationData.java

```diff
@@ -263,6 +274,151 @@ public int hashCode() {
+                                                logger.info(String.format("Property key: %s, List index: %d, a.value: %s, b.value: %s",
+                                                        propName, i, aList.get(i).getValue(), bList.get(i).getValue()));
+                                            }
+                                            return eq;
+                                        });
+                            } else if(aProp instanceof MapProperty && bProp instanceof MapProperty) {
+                                Map<String, SimpleProperty> aMap = ((MapProperty) aProp).getMap();
+                                Map<String, SimpleProperty> bMap = ((MapProperty) bProp).getMap();
+                                return aMap.keySet().equals(bMap.keySet()) &&
+                                        aMap.keySet().stream().allMatch(k -> {
+                                            boolean eq = simplePropertyDerefEquals(a, b, aMap.get(k), bMap.get(k), propName);
+                                            if(!eq) {
+                                                logger.info(String.format("Property key: %s, PropValue Key: %s, a.value: %s, b.value: %s",
+                                                        propName, k, aMap.get(k).getValue(), bMap.get(k).getValue()));
+                                            }
+                                            return eq;
+                                        });
+                            } else {
+                                logger.severe(String.format("Unrecognized Property type: %s with name: %s", aProp.getClass().getName(), propName));
+                                return false;
+                            }
+                        } else if((!aPropOpt.isPresent()) && (!bPropOpt.isPresent())) {
+                            return true; // both missing is a match
+                        } else {
+                            logger.info(String.format("Property key: %s, aOpt: %s, bOpt: %s", propName, aPropOpt, bPropOpt));
+                            return false;
+                        }
+                    });
+        }
+    }
+
+    /**
+     * Checks whether two ConfigurationData objects are 'structurally equal'. Two objects are structurally
+     * equal when their classNames are the same and when all of their properties are equal or, if those
+     * properties refer to another ConfigurationData by name, if all of their properties are structurally
+     * equal recursively.
+     *
+     * <p></p>
+     *
+     * {@code aName} should be the name of an element of {@code a} that is to be compared to {@code bName}
+     * in {@code b}. {@code a} and {@code b} should each contain all the ConfigurationData objects
+     * needed to instantiate the objects named by {@code aName} and {@code bName} respectively. Objects not
+     * instantiated by traversing children of {@code aName} and {@code bName} are ignored.
+     *
```

Comment:
Might be useful to note that this method is extremely chatty at the info level in the Javadoc when it finds discrepancies. Most other things in OLCUT tend to log at fine unless it's an actual issue, and that might be appropriate here.